internal/project/projectdata/packageindex.go

```diff
@@ -48,3 +77,70 @@ var packageIndexCLILoadError error
 func PackageIndexCLILoadError() error {
 	return packageIndexCLILoadError
 }
+
+var packageIndexPackages []PackageIndexData
+
+// PackageIndexPackages returns the slice of package data for the package index.
+func PackageIndexPackages() []PackageIndexData {
+	return packageIndexPackages
+}
+
+var packageIndexPlatforms []PackageIndexData
+
+// PackageIndexPlatforms returns the slice of platform data for the package index.
+func PackageIndexPlatforms() []PackageIndexData {
+	return packageIndexPlatforms
+}
+
+var packageIndexTools []PackageIndexData
+
+// PackageIndexTools returns the slice of tool data for the package index.
+func PackageIndexTools() []PackageIndexData {
+	return packageIndexTools
+}
+
+var packageIndexSystems []PackageIndexData
+
+// PackageIndexSystems returns the slice of system data for the package index.
+func PackageIndexSystems() []PackageIndexData {
+	return packageIndexSystems
+}
+
+func getPackageIndexData(interfaceObject map[string]interface{}, pointerPrefix string, dataKey string, iDPrefix string, iDKey string, versionKey string) []PackageIndexData {
+	var data []PackageIndexData
+
+	interfaceSlice, ok := interfaceObject[dataKey].([]interface{})
+	if ok {
+		for index, interfaceElement := range interfaceSlice {
+			object, ok := interfaceElement.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			var iD string
+			iDSuffix, ok := object[iDKey].(string)
+			if !ok {
+				continue
+			}
+			iD = iDPrefix + iDSuffix
+			if versionKey != "" {
+				iDVersion, ok := object[versionKey].(string)
+				if !ok {
+					continue
+				}
+				iD = iD + "@" + iDVersion
+			}
+
+			data = append(
+				data,
+				PackageIndexData{
+					ID:          iD,
+					JSONPointer: fmt.Sprintf("%s/%s/%v", pointerPrefix, dataKey, index),
+					Object:      object,
+				},
+			)
+		}
+	}
```

Comment:
I would have done like this to make it a bit more readable, but it works nonetheless so it's fine in any case, just a personal preference really.

```suggestion
	if !ok {
		return data
	}
	
	for index, interfaceElement := range interfaceSlice {
		object, ok := interfaceElement.(map[string]interface{})
		if !ok {
			continue
		}

		var iD string
		iDSuffix, ok := object[iDKey].(string)
		if !ok {
			continue
		}
		iD = iDPrefix + iDSuffix
		if versionKey != "" {
			iDVersion, ok := object[versionKey].(string)
			if !ok {
				continue
			}
			iD = iD + "@" + iDVersion
		}

		data = append(
			data,
			PackageIndexData{
				ID:          iD,
				JSONPointer: fmt.Sprintf("%s/%s/%v", pointerPrefix, dataKey, index),
				Object:      object,
			},
		)
	}
```

Reply:
Done:
https://github.com/arduino/arduino-lint/compare/649d6d4388da58f73eadf82fbf28e21865cfbbca..4921733f23d30a5c68d20353f763d177ab826505
Thanks Silvano!